source/common/network/address_impl.cc

```diff
@@ -133,18 +133,26 @@ addressFromSockAddrOrDie(const sockaddr_storage& ss, socklen_t ss_len, os_fd_t f
   return *address;
 }
 
+const std::string& IpInstance::asString() const {
+  if (address_port_string_.empty()) {
```

Comment:
I think you should delete this check. This is still racy if the member of the string that is checked for empty is set, but the data isn't yet set. I think the call_once check, in the already fully initialized state, should be pretty fast.

Reply:
I moved the check into call_once().